### PR TITLE
Speed up test suite via distributed testing

### PR DESCRIPTION
**Pull request by ichard26**
Since we now run the test suite twice, one with Python 2 and another
without, full test runs are getting pretty slow. Let's try to
fix that with parallization.

Also use verbose mode on CI since more logs is usually better since
getting more is quite literally impossible.

The main issue we'll face with this is we'll hit
pytest-dev/pytest-xdist#620 sometimes
(although pretty rarely). I suppose we can test this and see if how bad
this bug is for us, and revert if necessary down the line.

Also let's have some colours :tada:

---

Comparison:

After PR:

Screenshot from 2021-05-04 16-10-44
https://github.com/psf/black/actions/runs/811254652

Before PR:

Screenshot from 2021-05-04 16-12-25
https://github.com/psf/black/actions/runs/811220250

### CODE REVIEW

**Comment by ichard26** [comment]
I love how after the force push the [Test workflow](https://github.com/psf/black/actions/runs/811285467) takes 7 minutes to run, I swear there are improvements and this isn't useless :rofl:

**Comment by cooperlees** [comment]
> I _love_ how after the force push the [Test workflow](https://github.com/psf/black/actions/runs/811285467) takes 7 minutes to run, I swear there are improvements and this isn't useless

Compare it locally on a controlled platform (i.e. you box). That's enough evidence.

File: tox.ini [hunk 1]
```diff
@@ -9,9 +9,9 @@ deps =
 commands =
     pip install -e .[d]
     coverage erase
-    coverage run -m pytest tests -m "not python2" {posargs}
+    pytest tests -m "not python2" --numprocesses auto --cov src/ --cov tests/ {posargs}
```
Comment (cooperlees) [inline tox.ini:12]: pytest does not respect `.coveragerc` ?

**Review by cooperlees — CHANGES_REQUESTED** [review]
GitHub Actions gives us 1 CPU core, so that is why `auto` is not helping here:

From https://pypi.org/project/pytest-xdist/:
```
Pass -n auto to use as many processes as your computer has CPU cores.
```

We might need to just pass `-n 2` here?

The dream is `auto` for developers and something smarter for actions if doable.

File: tox.ini (hunk 1 above)
Comment (ichard26) [reply tox.ini:12]: I'm not actually sure, but good point, I'll check again.

**Comment by ichard26** [comment]
> GitHub Actions gives us 1 CPU core

This is false, but I see where you could get that understanding. The reason for my previous comment was because the GHA Test workflow run before actually saw performance improvements but the one after the forcepush didn't. This change does have improvements, but I was laughing because that workflow run wasn't exactly good at proving this change is beneficial. 

Anyway, in these logs you can tell the tests are running in parallel:

Linux: https://github.com/psf/black/pull/2196/checks?check_run_id=2503919997#step:6:60 
MacOS: https://github.com/psf/black/pull/2196/checks?check_run_id=2503920027#step:5:59
Windows: https://github.com/psf/black/pull/2196/checks?check_run_id=2503920051#step:5:59

Matter of fact, MacOS has three cores! These claims are also backed up the [GitHub Actions documentation](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources).

But I checked again locally and there are performance improvements:

`4m 33s 362ms` without this change and `3m 53s 547ms ` with this change for a `tox -e py38 -r` run.

File: tox.ini (hunk 1 above)
Comment (ichard26) [reply tox.ini]: Okay so `pytest-cov` does respect `.coveragerc`. I just assumed that `--cov` needed an argument so coverage would be collected but I was wrong. `pytest tests -m "not python2" --numprocesses auto --cov {posargs}` works just fine. Great catch! Thank you!

**Review by cooperlees — APPROVED** [review]
Sweet - Thanks for double checking. Seems 2 cores for linux/windows and 3 for Mac OS X reading docs.

**Comment by ichard26** [comment]
This \*should\* be ready to go now that I fixed the broken syntax in tox.ini (thanks Łukasz for fixing the conflict!):

Screenshot from 2021-05-07 10-35-39

I'm not merging my own PR though.

**Comment by Dwalinnewyork** [comment]
[AUTHORS.md](https://github.com/psf/black/files/6896889/AUTHORS.md)
[CHANGES.md](https://github.com/psf/black/files/6896890/CHANGES.md)
[CONTRIBUTING.md](https://github.com/psf/black/files/6896892/CONTRIBUTING.md)
[README.md](https://github.com/psf/black/files/6896893/README.md)
[test_requirements.txt](https://github.com/psf/black/files/6896894/test_requirements.txt)

**Comment by cooperlees** [comment]
@Dwalinnewyork - Why did you share this? What's your goal here?

**Pull request merged**